src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/CreateCollectionOperation.java

```diff
@@ -1,41 +1,134 @@
+      MISSING_KEYSPACE_FUNCTION =
+          keyspace -> {
+            String message =
+                "INVALID_ARGUMENT: Unknown namespace '%s', you must create it first."
+                    .formatted(keyspace);
+            Exception exception = new JsonApiException(ErrorCode.NAMESPACE_DOES_NOT_EXIST, message);
+            return Uni.createFrom().failure(exception);
+          };
+
   public static CreateCollectionOperation withVectorSearch(
       CommandContext commandContext,
+      ObjectMapper objectMapper,
+      SchemaManager schemaManager,
       String name,
       int vectorSize,
       String vectorFunction,
       String vectorize) {
     return new CreateCollectionOperation(
-        commandContext, name, true, vectorSize, vectorFunction, vectorize);
+        commandContext,
+        objectMapper,
+        schemaManager,
+        name,
+        true,
+        vectorSize,
+        vectorFunction,
+        vectorize);
   }
 
   public static CreateCollectionOperation withoutVectorSearch(
-      CommandContext commandContext, String name) {
-    return new CreateCollectionOperation(commandContext, name, false, 0, null, null);
+      CommandContext commandContext,
+      ObjectMapper objectMapper,
+      SchemaManager schemaManager,
+      String name) {
+    return new CreateCollectionOperation(
+        commandContext, objectMapper, schemaManager, name, false, 0, null, null);
   }
 
   @Override
   public Uni<Supplier<CommandResult>> execute(QueryExecutor queryExecutor) {
+    return schemaManager
+        .getTable(commandContext.namespace(), name, MISSING_KEYSPACE_FUNCTION)
+        .onItem()
+        .transformToUni(
+            table -> {
+              // table doesn't exist
+              if (table == null) {
+                return executeCollectionCreation(queryExecutor);
+              }
+              // get collection settings from the existing collection
+              CollectionSettings collectionSettings =
+                  CollectionSettings.getCollectionSettings(table, objectMapper);
+              // get collection settings from user input
+              CollectionSettings collectionSettings_cur =
+                  CollectionSettings.getCollectionSettings(
+                      name,
+                      vectorSearch,
+                      vectorSize,
+                      CollectionSettings.SimilarityFunction.fromString(vectorFunction),
+                      vectorize,
+                      objectMapper);
+              // if table exists and user want to create a vector collection with the same name
+              if (vectorSearch) {
+                // if existing collection is a vector collection
+                if (collectionSettings.vectorEnabled()) {
+                  if (collectionSettings.equals(collectionSettings_cur)) {
+                    // if settings are equal, no error
+                    return executeCollectionCreation(queryExecutor);
```

Comment:
Fine not to error out, but can't we just return ok/good at this point since no creation is needed? (table exists with same config).
Is there a query to be executed? Or is `executeCollectionCreation()` just misleading name and it will do nothing?

Reply:
The query executed in `executeCollectionCreation()` is `CREATE TABLE IF NOT EXISTS`. Since the table already exists, the database will do nothing and return 'ok', and we will return 'ok' status to the user as well. Would you prefer that we directly return 'ok' here instead of interacting with the database? 

I am not sure about this situation: If the database is experiencing issues, bypassing the database interaction would prevent us from receiving any error messages. This could create a misleading situation where the user believes they have successfully created a collection, but may encounter problems when trying to interact with it later.

Reply:
Good question. But when attempting to use "CREATE TABLE IF NOT EXISTS", I am not sure interaction occurs, beyond checking DB exists (which we just did).
So in that sense I think it would be best to just return ok directly.

@maheshrajamani @amorton WDYT?

This is something we can discuss on team daily meeting, to resolve it quickly.

Reply:
@tatu-at-datastax Wanted to run it again so in case any index is missed it will be created. This can happen if index failed because of guardrail and user decides to drop a different collection and create this again.